### PR TITLE
Changes to support ebpf and ipv6

### PR DESCRIPTION
**Pull request by sridhartigera**
## Description

Remove the check to disallow eBPF and IPv6.

## For PR author

- [x] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

### CODE REVIEW

**Review by tomastigera — COMMENTED** [review]
LGTM but I will let operator-maintainers to approve :)

File: pkg/controller/installation/validation.go [hunk 1]
```diff
@@ -291,8 +292,16 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			}
 		}
 
-		if bpfDataplane && instance.Spec.CalicoNetwork.NodeAddressAutodetectionV4 == nil {
-			return fmt.Errorf("spec.calicoNetwork.nodeAddressAutodetectionV4 is required for the BPF dataplane")
+		if bpfDataplane {
+			if bpfIPv6Enabled {
```
Comment (tmjd) [inline pkg/controller/installation/validation.go:296]: Is dual stack supported with bpf? If not then I think we should have a check for that.
I think I'd also suggest moving these validations to under the `if v4pool != nil {` and `if v6pool != nil {` blocks.

Comment (sridhartigera) [reply pkg/controller/installation/validation.go]: Dual stack is not supported. Added a check for that.

**Review by tmjd — APPROVED** [review]
LGTM

**Pull request merged**